rust/azure_iot_operations_mqtt/src/connection_settings.rs

```diff
@@ -393,6 +393,13 @@ mod tests {
             .hostname("test_host".to_string())
             .build();
         assert!(connection_settings_builder_result.is_ok());
+
+        let connection_settings = connection_settings_builder_result.unwrap();
+        assert_eq!(connection_settings.client_id().as_str(), "test_client_id");
```

Comment:
I think this is an unnecessary addition that distracts from the actual test. This test is about the builder validation, not the actual settings object, so the getters are immaterial to the unit under test.

If you want to have a getter test I would prefer it be separate (although I think it's rather unnecessary due to the derivation) or consistently applied to all tests (since we would want to show the getters work for all values, not just the minimum required values by the builder).

Reply:
If you don't think the explicit test is necessary, I'm more inclined to just remove it and rely on the macro.